insonmnia/worker/server.go

```diff
@@ -371,18 +372,38 @@ func (m *Worker) cancelDealTasks(deal *pb.Deal) error {
 }
 
 type runningContainerInfo struct {
-	Description Description   `json:"description,omitempty"`
-	Cinfo       ContainerInfo `json:"cinfo,omitempty"`
-	Spec        pb.TaskSpec   `json:"spec,omitempty"`
+	Description descriptionMarshallable   `json:"description,omitempty"`
+	Cinfo       containerInfoMarshallable `json:"cinfo,omitempty"`
+	Spec        pb.TaskSpec               `json:"spec,omitempty"`
+}
+
+type descriptionMarshallable struct {
+	Description
+	Reference reference.Field `json:"Reference"`
+	Networks  []*structs.NetworkSpec
+}
+
+type containerInfoMarshallable struct {
```

Comment:
why can't we introduce wrapper for Public key instead of containerInfo?

Reply:
It is a matter of taste: some ugly wrapping and unwrapping will be done in any case. But when adding more fields to container info that have to have this ugly marshalling, you wouldn't have to add more new wrapper types for those fields.